src/sciline/pipeline.py

```diff
@@ -200,6 +209,15 @@ def _repr_html_(self) -> str:
         nodes = ((key, data) for key, data in self.underlying_graph.nodes.items())
         return pipeline_html_repr(nodes)
 
+    def final_result_keys(self) -> tuple[type, ...]:
+        """Returns the keys that are not inputs to any other providers."""
+        sink_nodes = [
+            cast(type, node)
```

Comment:
Not sure I understand. What is this `cast` doing?

Reply:
It's just for appeasing mypy.
We promise the return type of the function is `tuple[type, ...]` because we know our keys are types.
But what we get from the `underlying_graph` is `Any`.

Reply:
Okay took another look now and it was actually not necessary to cast in case we just specified the return as `tuple[Key, ...]` instead

Reply:
Indeed. Our keys are, in general not `type`.

Reply:
We are using `type` in a lot of other places instead of `Key` though, that's what confused me.
For example https://github.com/scipp/sciline/blob/main/src/sciline/pipeline.py#L108-L111

Is that just a mistake or is there a difference between the contexts that makes `type` the right annotation there?